packages/edit-widgets/src/blocks/widget-area/edit/index.js

```diff
@@ -20,36 +21,47 @@ export default function WidgetAreaEdit( {
 	className,
 	attributes: { id, name },
 } ) {
-	const isOpen = useSelect(
-		( select ) =>
-			select( 'core/edit-widgets' ).getIsWidgetAreaOpen( clientId ),
+	const { isOpen, isDraggingBlocks } = useSelect(
+		( select ) => ( {
+			isOpen: select( 'core/edit-widgets' ).getIsWidgetAreaOpen(
+				clientId
+			),
+			isDraggingBlocks: select( 'core/block-editor' ).isDraggingBlocks(),
+		} ),
 		[ clientId ]
 	);
 	const { setIsWidgetAreaOpen } = useDispatch( 'core/edit-widgets' );
+	const openOnDragOver = useCallback( () => {
+		if ( ! isOpen && isDraggingBlocks ) {
+			setIsWidgetAreaOpen( clientId, true );
+		}
+	}, [ clientId, isOpen, setIsWidgetAreaOpen, isDraggingBlocks ] );
 
 	return (
-		<Panel className={ className }>
-			<PanelBody
-				title={ name }
-				opened={ isOpen }
-				onToggle={ () => {
-					setIsWidgetAreaOpen( clientId, ! isOpen );
-				} }
-			>
-				{ ( { opened } ) => (
-					// This is required to ensure LegacyWidget blocks are not unmounted when the panel is collapsed.
-					// Unmounting legacy widgets may have unintended consequences (e.g. TinyMCE not being properly reinitialized)
-					<DisclosureContent visible={ opened }>
-						<EntityProvider
-							kind="root"
-							type="postType"
-							id={ `widget-area-${ id }` }
-						>
-							<WidgetAreaInnerBlocks />
-						</EntityProvider>
-					</DisclosureContent>
-				) }
-			</PanelBody>
-		</Panel>
+		<div onDragOver={ openOnDragOver }>
```

Comment:
An alternative would be to do `<Panel onDragOver={ openOnDragOver }>` and then update `<Panel />` to apply arbitrary properties to its root element, thoughts?

Reply:
Yep, I think this is ok.

Another alternative would be `<Panel openOnDragOver>` making the behavior implicit to Panel. Though I think given PanelBody is more of a controlled component it's better to keep it explicit like you have it here.

Reply:
Actually, I realise there's some block CSS that might need to be adjusted (though it's currently broken anyway)